### PR TITLE
MSFT:15871527 Add macros to convert OOM to HRESULT on OnDebuggerDetached.

### PR DESCRIPTION
**Pull request by atulkatti**

### CODE REVIEW

**Comment by atulkatti** [comment]
@dotnet-bot test OSX _no_jit_shared_osx_osx_test please
@dotnet-bot test OSX static_osx_osx_debug please

File: lib/Runtime/Base/ScriptContext.cpp [hunk 1]
```diff
@@ -3716,9 +3716,21 @@ namespace Js
         // Debugger attach/detach failure is catastrophic, take down the process
         DEBUGGER_ATTACHDETACH_FATAL_ERROR_IF_FAILED(hr);
 
+        HRESULT hrEntryPointUpdate = S_OK;
+        BEGIN_TRANSLATE_OOM_TO_HRESULT_NESTED
```
Comment (dilijev) [inline lib/Runtime/Base/ScriptContext.cpp]: nit: other usages of this macro pair wrap the body with an open brace and close brace to make the scope more obvious

**Comment by atulkatti** [comment]
@dotnet-bot test Windows 10 ci_slow_x64_debug please

**Comment by atulkatti** [comment]
@dotnet-bot test Windows 10 ci_slow_x64_debug please
@dotnet-bot test Windows 10 x64_release please

**Comment by atulkatti** [comment]
@dotnet-bot test Windows 10 x64_release please

**Pull request merged**